### ISSUE TITLE
C# Script Templates being parsed during build

### ISSUE THREAD

**Issue by duarteroso** [godotengine/godot#60703]
### Godot version

3.4.4.stable

### System information

Windows 10, macOS

### Issue description

Project parses the contents of `script_templates` when pressing `Build`

### Steps to reproduce

1) Add a folder named `script_templates` 
2) Add a template `game_template.cs` inside that folder 
```cs
using Godot;

namespace Game
{
    public sealed class %CLASS% : %BASE%
    {
    }
}
```
3) Add a ignore `.gdignore` inside that folder
4) Play the game, notice that no error is thrown
5) Go to `Project > Tools > C#` and select the option `Create C# solution`
6) Build the project, notice that the `game_template.cs` will be parsed and errors will be thrown

### Minimal reproduction project

[templates.zip](https://github.com/godotengine/godot/files/8601761/templates.zip)

**Comment by Delpire** [comment]
I'm not sure this is a bug. I believe Godot now uses the new style SDK csproj format which automatically searches the directory for CS files. Rather than excluding the file in the .gdignore (or perhaps in addition to), you want to ignore the file in the csproj as well. 

https://stackoverflow.com/questions/43173811/how-do-i-exclude-files-folders-from-a-net-core-standard-project

**Comment by Calinou** [comment]
> Rather than excluding the file in the .gdignore (or perhaps in addition to), you want to ignore the file in the csproj as well.

It makes sense for the Godot editor to automatically ignore `script_templates/*.cs` within the project folder.

**Comment by paulloz** [comment]
> It makes sense for the Godot editor to automatically ignore `script_templates/*.cs` within the project folder. This could be done while generating the `.csproj` file.

Fair. But since that path is configurable in the project settings, invalidation becomes a bit of a nightmare.

**Comment by duarteroso** [comment]
Any workaround I could apply in the mean time?

**Comment by raulsntos** [comment]
> Any workaround I could apply in the mean time?

Yes, take a look at the [_Exclude files from the build_](https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-exclude-files-from-the-build) documentation page from Microsoft.
